Cargo.toml

```diff
@@ -10,19 +10,29 @@ license-file = "LICENSE"
 description = "Library to facilitate DeFi on Liquid"
 
 [dependencies]
+aes-gcm-siv = { version = "0.9", features = ["std"] }
```

Comment:
:thinking: I still think we should move the `wallet` (basically almost everything new that comes with this PR) to a different crate within the workspace. 

Our elements wallet shouldn't need to know about our protocols, and it actually doesn't if you look at the `use` statements in `wallet.rs`. I think by keeping wallet and protocols separate we can more clearly see the relationship between the two parts: the wallet is used to run the protocol, but without the protocol knowing about the specific wallet you use.

Reply:
One downside I do see with the separation is the need to publish another `elements-wallet` crate to `crates.io`, which must be updated in sync with `baru`. Perhaps that overhead is too much for what we gain.

Reply:
I would hesitate to publish another crate for now. At least until you have PSETs. Until then, I think you will find yourself editing the protocol and the wallet together to provide interfaces that go well with each other.

Reply:
You can already maintain a nice separation by having a `wallet` sub-module that owns all of its module like `coin_selection` etc.